libvast/src/error.cpp

```diff
@@ -51,8 +51,12 @@ const char* descriptions[] = {
   "missing_component",
   "unimplemented",
   "silent",
+  "out_of_memory",
 };
 
+static_assert(ec{sizeof(descriptions) / sizeof(*descriptions)} == ec::ec_count,
+              "Mismatch between number of error codes and descriptions");
```

Comment:
```suggestion
static_assert(ec{std::size(descriptions)} == ec::ec_count,
              "Mismatch between number of error codes and descriptions");
```

I think this is more idiomatic for modern C++, and also more resilient in case we switch to `std::array` in the future.